scripts/sanitycheck

```diff
@@ -3342,6 +3347,12 @@ structure in the main Zephyr tree: boards/<arch>/<board_name>/""")
         NOTE: west-flash must be enabled to use this option.
         """
     )
+    parser.add_argument(
+        "--valgrind", action="store_true",
+	help="""Run binary through valgrind and check for memory leaks.
```

Comment:
Better rephrase a bit to:
"and check for memory leaks." => "and check for several memory access errors"

---------------------

Background: you won't catch Zephyr leaks with it, but you do catch other issues:
* Accessing memory you shouldn't, e.g. overrunning and underrunning heap blocks, overrunning the top of the stack, and accessing memory after it has been freed.
* Using undefined values, i.e. values that have not been initialised, or that have been derived from other undefined values.
* Overlapping src and dst pointers in memcpy and related functions.
* Passing a fishy (presumably negative) value to the size parameter of a memory allocation function.